### ISSUE TITLE
About global_scale parameter values

### ISSUE THREAD

**Issue by aqibsaeed** [mary-phuong/multiexit-distillation#5]
Hi Phuong,

Could you please explain what is global_scale parameter is and how did you choose the particular values i.e, 2.0 and 5 ? in global_scale = 2.0 * 5/cf_net['n_exits']

**Comment by mary-phuong** [comment]
Hi, 

it's a scalar multiplier for the entire loss. If you set it to 3.0, the loss will be three times as large as when you set it to 1.0.

The value 2.0 was chosen because it worked well for my ImageNet network with five exits.
For a different number of exits, I modify the multiplier proportionally to 1/n_exits. This is simply because the loss is a sum of n_exits terms, and it seems natural to take the average per-exit loss rather than the sum.

**Comment by aqibsaeed** [comment]
Thank you! Makes sense.

**Comment by aqibsaeed** [comment]
Another question just to clarify the total loss (dist_loss and cross_ent). As you mentioned taking average of per-exit loss. Does it mean that if there are 5 exits then we will have 5 (dist_loss and cross_ent) losses per exit averaged over a batch let's say dist_loss = [1,2,3,4,5] and cross_ent = [6,7,8,9,10]. So the average in this case dist_loss = 3 and cross_ent = 8, right? and total_loss will be global_scale * (3+8) or is it just 3+8?

**Comment by mary-phuong** [comment]
There is also the constant C (between 0 and 1) that is the relative importance of cross_ent vs. dist_loss, and the distillation temperature T. In the example you give:

total_loss = global_scale * ( (C*T^2) * 3 + (1-C) * 8).

([Here](https://github.com/mary-phuong/multiexit-distillation/blob/master/main.py#L679)'s the code.)

**Comment by aqibsaeed** [comment]
Thank you Mary.
